Scenes/Monsters/Bat.gd

```diff
@@ -1,26 +1,33 @@
 extends KinematicBody2D
 
-#var knockback = Vector2.ZERO
+const EnemyDeathEffect = preload("res://newEffects/BatEffect.tscn")
 
-#func _physics_process(delta):
-	#knockback = knockback.move_toward(Vector2.ZERO, 200 * delta)
-	#knockback = move_and_slide(knockback)
-	
+var knockback = Vector2.ZERO
 
-# Declare member variables here. Examples:
-# var a = 2
-# var b = "text"
 
+onready var stats = $Stats
 
-# Called when the node enters the scene tree for the first time.
+#func _ready():
+	#print(stats.max_health)
+	#print(stats.health)
 
-# Called every frame. 'delta' is the elapsed time since the previous frame.
-#func _process(delta):
-#	pass
+func _physics_process(delta):
+	knockback = knockback.move_toward(Vector2.ZERO, 200 * delta)
+	knockback = move_and_slide(knockback)
 
 
 func _on_Hurtbox_area_entered(area):
-	#knockback = area.knockback_vector * 120
-	queue_free()
+	stats.health -= 1
+	
+	knockback = Vector2.RIGHT * 120
```

Comment:
I guess here we need to compute the direction vector. If my math is not that rusty, a (area.global_position - global_position).normalize() would do the trick

Reply:
Might need to update the order depending if this is applying a knockback to the player or the bat :D